census/vanilla-tf/trainer/task.py

```diff
@@ -44,228 +43,149 @@
-  """Sparse feature cross of the feature SparseTensors."""
-  return sparse_feature_cross_op.sparse_feature_cross(
-      feature_tensors,
-      hashed_output=True,
-      num_buckets=num_buckets,
-      hash_key=tf.contrib.layers.SPARSE_FEATURE_CROSS_DEFAULT_HASH_KEY,
-      name=name
-  )
-
-def to_continuous_and_categorical(features, labels, batch_size):
-  """Convert the input columns to continuous and sparse tensor and
-     labels to the one hot tensor.
-  """
-
-  # get the continuous columns
-  continuous_columns = [
-      features.get(col) for col in CONTINUOUS_COLS
-  ]
-
-  # convert the categorical columns into sparse tensors
-  categorical_columns = [
-      tf.SparseTensor(
-          indices=[[i, 0] for i in range(batch_size)],
-          values=string_ops.string_to_hash_bucket_fast(features[col], vocab),
-          dense_shape=[batch_size, 1])
-      for col, vocab in CATEGORICAL_COLS
-  ]
-
-  # convert the labels into one hot encoding
-  one_hot_labels = tf.one_hot(labels, 2)
-
-  return continuous_columns + categorical_columns, one_hot_labels
-
-
-def generate_wide_columns(input_columns):
-  """Generate wide columns by adding feature crosses of SparseTensors."""
-  (age, education_num, capital_gain, capital_loss, hours_per_week,
-   gender, race, education, marital_status, relationship, workclass,
-   occupation, native_country) = input_columns
-
-  # bucketize the age feature
-  # See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/ops/bucketization_op.py
-  age_bucket = bucketization_op.bucketize(age, [18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
-
-  wide_columns = [
-      (sparse_cross([education, occupation], 15 * 16, name='edu_occ'), 15 * 16),
-      (gender, 2),
-      (native_country, 42),
-      (education, 16),
-      (occupation, 15),
-      (workclass, 9),
-      (marital_status, 7),
-      (relationship, 6)
-  ]
-
-  dense_tensors = [sparse_to_dense(col[0], col[1]) for col in wide_columns]
-
-  return tf.concat(dense_tensors, 1)
-
-
+        tf.nn.sparse_softmax_cross_entropy_with_logits(
+            logits=logits, labels=tf.squeeze(labels)))
+
+  train_op = tf.train.GradientDescentOptimizer(learning_rate).minimize(
+        cross_entropy, global_step=global_step)
+
+  probabilities = tf.nn.softmax(logits)
+  predictions = tf.argmax(probabilities, 1)
+  accuracy_op, update_acc = tf.contrib.metrics.streaming_accuracy(predictions, labels)
+
+  return train_op, accuracy_op, update_acc, global_step, predictions
+
+
+def convert_sparse_columns(features):
+  for col, bucket_size in HASH_BUCKET_COLS:
+    features[col] = string_ops.string_to_hash_bucket_fast(
+            features[col], bucket_size)
+  
+  for col, keys in KEY_COLS:
+    table = tf.contrib.lookup.string_to_index_table_from_tensor(
+        tf.constant(keys))
+    features[col] = table.lookup(features[col])
+  
+  for col, size in CATEGORICAL_COLS:
+    features[col] = tf.squeeze(tf.one_hot(
```

Comment:
Concat requires Tensors all of the same rank, and we don't transform the real valued columns.